### PR TITLE
[Popover] Multiple small enhancements

### LINKED ISSUE #588: [Popover] Multiple small enhancements
Required by CraveFood/webapp#3141.

- [X] Add `onOpen` and `onClose` props
- [x] `trigger` prop: accept a function, so we can pass `isVisible`. Keep accepting regular nodes, for backward compatibility
- [X] Add `width` prop, so it works with fluid buttons as trigger
- [x] Expose `hideArrow` from tooltip. Default: `false` for backward compatibility

### PR DESCRIPTION
**Pull request by vNakamura**
## Closes Issues

* Closes #588 

### Reviewer 1

* [ ] Functions properly
* [ ] Good readability, clarity and overall quality
* [ ] Has tests
* [ ] Has docs

### Reviewer 2

* [ ] Functions properly
* [ ] Good readability, clarity and overall quality
* [ ] Has tests
* [ ] Has docs

### CODE REVIEW

File: packages/popover/src/Popover.test.js [hunk 1]
```diff
@@ -174,4 +174,134 @@ describe("Popover", () => {
...
+      const wrapper = shallow(
+        <Popover
+          trigger={<span>trigger</span>}
+          content={() => <div>content</div>}
+        />
+      );
+
+      expect(() => {
+        wrapper.find("#trigger").simulate("click");
+      }).not.toThrowError();
```
Comment (alcferreira) [inline packages/popover/src/Popover.test.js]: 👍

**Comment by vNakamura** [comment]
I'm changing the issue to group multiple enhancements. So this PR is back to WIP

File: packages/popover/README.md [hunk 2]
```diff
@@ -34,6 +34,8 @@ npm install @crave/farmblocks-popover
 | zIndex   | zIndex of popover                                                  | number                                     |          |         |
 | padding  | padding of popover contents                                        | string                                     |          | 4px     |
 | overflow | overflow of popover contents                                       | string                                     |          |         |
+| onOpen   | called when the popver opens                                       | func                                       |          |         |
```
Comment (fgameiro) [inline packages/popover/README.md]: Minor, typo in `popover`

File: packages/popover/README.md [hunk 3]
```diff
@@ -34,6 +34,8 @@ npm install @crave/farmblocks-popover
...
+| onClose  | called when thr popover closes                                     | func                                       |          |         |
```
Comment (fgameiro) [inline packages/popover/README.md]: Typo in `the`

File: packages/popover/src/Popover.js [hunk 4]
```diff
@@ -62,13 +79,22 @@ class Popover extends React.Component {
...
     align: PropTypes.oneOf(values(alignments)),
     zIndex: PropTypes.number,
     padding: PropTypes.string,
     overflow: PropTypes.string,
-    onOutsideClick: PropTypes.func
+    triggerWidth: PropTypes.string,
+    onOutsideClick: PropTypes.func,
+    onOpen: PropTypes.func,
+    onClose: PropTypes.func,
+    hideArrow: PropTypes.bool
```
Comment (alcferreira) [inline packages/popover/src/Popover.js]: What about `hideTooltipArrow`?

Comment (alcferreira) [reply packages/popover/src/Popover.js]: Or even `showTooltipArrow` since the default is to hide the arrow...

**Pull request merged**